src/psyclone/tests/psyir/transformations/inline_trans_test.py

```diff
@@ -1621,9 +1621,39 @@ def test_validate_codeblock(fortran_reader):
             "cannot be inlined" in str(err.value))
 
 
-def test_validate_saved_var(fortran_reader):
+def test_validate_unknowntype_var(fortran_reader):
     '''
-    Test that validate rejects a subroutine with a 'save'd variable.
+    Test that validate rejects a subroutine with UnknownType variables.
+
+    '''
+    code = (
+        "module test_mod\n"
+        "contains\n"
+        "subroutine main\n"
+        "  real, target :: var = 0.0\n"
+        "  real, pointer :: ptr => null()\n"
+        "  ptr => var\n"
+        "  call sub(ptr)\n"
+        "end subroutine main\n"
+        "subroutine sub(x)\n"
+        "  real, pointer, intent(inout) :: x\n"
+        "  x = x + 1.0\n"
+        "end subroutine sub\n"
+        "end module test_mod\n"
+    )
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    with pytest.raises(TransformationError) as err:
+        inline_trans.validate(routine)
+    assert ("Routine 'sub' cannot be inlined because it contains a Symbol "
```

Comment:
As commented earlier, I can imagine that a user might sometimes want to ignore this warning. However, that can be a separate PR.

Reply:
I have relaxed this a bit to accept UnknownTypes unless they have ArgumentInterface (because we don't know if assignment is valid) or UnknownInterface (because we don't know the interface implications of changing the symbol to another scope)

Reply:
Great.